etc/kernel-launchers/python/scripts/launch_ipykernel.py

```diff
@@ -360,7 +364,64 @@ def get_server_request(sock):
     return request_info
 
 
-def server_listener(sock, parent_pid):
+def get_application_id(ui_port):
+    application_id = ""
+    url = f"http://localhost:{ui_port}/api/v1/applications/"
```

Comment:
Should this be actually based on the Spark master URL rather than assuming always localhost? Maybe instead of using an env var for the port, we should just propagate the master URL?

Reply:
When you say "propagate the master URL" you mean share it with the server listener process? I tried sharing ui_port by setting it during spark context initialisation and sharing with manager.value but that didn't work.

So do you have any alternative for this?

Reply:
What I am saying is, instead of providing a env var with just the master port, use one for the whole spark master unless that is available somewhere else.